osu.Game/Screens/Play/Player.cs

```diff
@@ -884,6 +886,16 @@ public override void OnEntering(IScreen last)
             if (!LoadedBeatmapSuccessfully)
                 return;
 
+            hasBackgroundToFit = !Beatmap.Value.Storyboard.ReplacesBackground && Beatmap.Value.Storyboard.HasDrawable;
+
+            if (hasBackgroundToFit)
```

Comment:
I would like to avoid needlessly inflating `Player` with yet another responsibility on top of what it already handles. I believe this logic should be implemented somewhere closer to the actual storyboard components if at all possible.

Reply:
It's possible that the background sprite can be put onto beatmap storyboard as an actual storyboard element. This approach doesn't struggle with the background scalings and doesn't need to change `BeatmapBackground `or involve `Player`. I initially started with it and it does has some problems. If you can confirm that it's a better approach i'll go with it.